googletest/src/gtest_main.cc

```diff
@@ -30,13 +30,20 @@
 #include <cstdio>
 #include "gtest/gtest.h"
 
-#ifdef ARDUINO
+#if GTEST_OS_ESP8266 || GTEST_OS_ESP32
+# if GTEST_OS_ESP8266
+extern "C" {
```

Comment:
I'm a bit unsure about this.
Could you explain why it is necessary?

Reply:
This was a weird thing.  The linked on esp8266 want these to be extern C. Other platforms I’ve used do not required for Arduino setup/loop functions. 

I am definitely open to a better way.